android/src/main/kotlin/com/baseflow/flutter_instance_manager/FlutterInstanceManagerPlugin.kt

```diff
@@ -0,0 +1,35 @@
+package com.baseflow.flutter_instance_manager
+
+import androidx.annotation.NonNull
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+/** FlutterInstanceManagerPlugin */
+class FlutterInstanceManagerPlugin: FlutterPlugin, MethodCallHandler {
+  /// The MethodChannel that will the communication between Flutter and native Android
+  ///
+  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
+  /// when the Flutter Engine is detached from the Activity
+  private lateinit var channel : MethodChannel
+
+  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_instance_manager")
+    channel.setMethodCallHandler(this)
```

Comment:
This can be a no-op (see also the implementation off [flutter_android_life_cycle](https://github.com/flutter/packages/blob/main/packages/flutter_plugin_android_lifecycle/android/src/main/java/io/flutter/plugins/flutter_plugin_android_lifecycle/FlutterAndroidLifecyclePlugin.java) plugin)